network-api/networkapi/wagtailpages/tests/buyersguide/test_editorial_content_index.py

```diff
@@ -156,15 +157,52 @@ def test_serve_paginated_items_page_2(self):
                 articles[self.items_per_page :],
             )
 
-    def test_items_route_exists(self):
-        route = self.content_index.reverse_subpage("items")
+    def test_serve_paginated_items_expanded_page_2_of_3(self):
+        """
+        Render all the items from the first and second page.
+
+        We don't want to show pages from the following pages.
+        Naviation to the first page should not be possible, because that page is
+        already included. Navigation to the next page should be possible.
+        """
+        page = 2
+        with self.setup_content_index_with_pages_of_children(pages=page + 1) as articles:
+
+            response = self.client.get(
+                self.content_index.url,
+                data={"page": page, "expanded": "true"},
+            )
+
+            index_of_first_not_expected_article = page * self.items_per_page
+            self.assertQuerysetEqual(
+                response.context["items"],
+                articles[:index_of_first_not_expected_article],
+            )
+            self.assertTrue(response.context["items"].has_next())
+            self.assertFalse(response.context["items"].has_previous())
+
+    def test_serve_paginated_items_expanded_page_2_of_2(self):
+        """There should be no other pages because they are all included."""
+        page = 2
+        with self.setup_content_index_with_pages_of_children(pages=page):
+
+            response = self.client.get(
+                self.content_index.url,
+                data={"page": page, "expanded": "true"},
+            )
 
-        self.assertEqual(route, "items/")
+            self.assertFalse(response.context["items"].has_next())
+            self.assertFalse(response.context["items"].has_previous())
 
-    def test_items_route_template(self):
-        url = self.get_items_route_url()
+    def test_hx_request_templates(self):
+        """
+        Only the items template (not the full page) template is used for HTMX requests.
```

Comment:
**Question**

I think the comment is throwing me a bit here, but does this test need updating to check the template fragment rather than the editorial_content_index_page ?